tests/installation/validate_no_self_update.pm

```diff
@@ -0,0 +1,34 @@
+# Copyright © 2020 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary: Validate installer self update is not attempted when explicitly disabled
+# Maintainer: QA SLE YaST <qa-sle-yast@suse.de>
+
+use base 'y2_installbase';
+use strict;
+use warnings;
+use testapi;
+
+sub run {
+    my $self = shift;
+    select_console('install-shell');
+    assert_script_run('test -z "$(ls -A /download)"',
+        fail_message => '/download directory is NOT empty, expected to be empty as no updates should be downloaded');
+    assert_script_run('! grep /var/log/YaST2/y2log -e "Trying installer update"',
```

Comment:
Can i ask you to try the usage of --files-without-match. i kinda dislike the use of `!` when there is another way, so it makes the expression more readable. For instance `grep /var/log/YaST2/y2log --files-without-match -e "Trying installer update"` it greps what files do not have the expression "..." and returns `0` if file found have it.

Reply:
Hmm, I need to check how exactly that flag works, because the idea is that finds files that do not contain this string, so I'm not 100% sure if it will return `0` in case no files found.

Reply:
Somehow `grep -L 'Trying installer update' /var/log/YaST2/y2log` gives `1` as a return code, even though it lists `/var/log/YaST2/y2log` as file without that string. Feels like a bug to me.

Reply:
![image](https://user-images.githubusercontent.com/28305643/91697818-07548480-eb72-11ea-9ded-612c1a8cf37d.png)

Reply:
So the only thing I can propose here is to rewrite this line with `script_run` and compare return code in perl code directly. Do you think this is better than negation of the bash command?

Reply:
you still need the `-e` flag. Can you try?

Reply:
The command I've used is equivalent to `grep /var/log/YaST2/y2log -L -e 'BLA' ` so result is same, and as you can see on the screenshot it works just fine and returns the file, as it doesn't contain the string. I will check if that's expected, and file the bug against grep.